--- acpica-unix-20110527/compiler/aslcompiler.h.orig    2011-05-27
17:25:04.000000000 +0200
+++ acpica-unix-20110527/compiler/aslcompiler.h 2011-05-27 21:47:40.338767397
+0200
@@ -361,7 +361,7 @@

 int
 AslCompilererror(
-    char                    *s);
+    const char                    *s);

 void
 AslCommonError (
--- acpica-unix-20110527/compiler/aslerror.c.orig       2011-05-27
17:25:04.000000000 +0200
+++ acpica-unix-20110527/compiler/aslerror.c    2011-05-27 21:47:26.495147836
+0200
@@ -692,7 +692,7 @@

 int
 AslCompilererror (
-    char                    *CompilerMessage)
+    const char                    *CompilerMessage)
 {

     AslCommonError (ASL_ERROR, ASL_MSG_SYNTAX, Gbl_CurrentLineNumber,