diff --git i/include/freetype/config/ftoption.h w/include/freetype/config/ftoption.h
index fe2ba15..22ce1a3 100644
--- i/include/freetype/config/ftoption.h
+++ w/include/freetype/config/ftoption.h
@@ -126,7 +126,7 @@
    * macro is not defined, FreeType offers alternative LCD rendering
    * technology that produces excellent output without LCD filtering.
    */
-/* #define FT_CONFIG_OPTION_SUBPIXEL_RENDERING */
+#define FT_CONFIG_OPTION_SUBPIXEL_RENDERING


   /**************************************************************************


