--- a/scripts/libmakepkg/tidy/strip.sh.in	2017-04-29 01:18:01.000000000 +0200
+++ b/scripts/libmakepkg/tidy/strip.sh.in	2018-04-21 15:22:09.126286998 +0200
@@ -108,6 +108,17 @@
 					esac;;
 				*application/x-executable*) # Binaries
 					strip_flags="$STRIP_BINARIES";;
+				*application/x-pie-executable*) ## file 5.33++
+					case "$binary" in
+						*.so*)
+							 msg2 "$(gettext "Stripping unneeded symbols from pie libraries...")"
+							strip_flags="$STRIP_SHARED";;
+						*.a)
+							strip_flags="$STRIP_STATIC";; ## possible with glibc 2.27+gcc8.x
+						*)
+							msg2 "$(gettext "Stripping unneeded symbols from pie binaries...")"
+							strip_flags="$STRIP_BINARIES";;
+					esac;;
 				*)
 					continue ;;
 			esac
