--- a/src/rpcb_svc_com.c	2015-04-27 10:07:43.000000000 -0400
+++ b/src/rpcb_svc_com.c	2015-11-01 13:52:11.668089322 -0500
@@ -1274,10 +1274,8 @@
 	a.rmt_localvers = fi->versnum;
 
 	xprt_set_caller(xprt, fi);
-	xprt->xp_auth = &svc_auth_none;
+	SVC_XP_AUTH(xprt) = svc_auth_none;
 	svc_sendreply(xprt, (xdrproc_t) xdr_rmtcall_result, (char *) &a);
-	SVCAUTH_DESTROY(xprt->xp_auth);
-	xprt->xp_auth = NULL;
 done:
 	if (buffer)
 		free(buffer);
