--- repo-clean.cpp	2011-06-25 21:41:51.434534186 +0200
+++ repo-clean.cpp.new	2011-06-25 21:45:22.885284393 +0200
@@ -621,7 +621,7 @@
                     string_list sublist = build_pkg_list(source_directory + "/" + dir_iter->leaf(),
                             recursive);
 #else
-                    string_list sublist = build_pkg_list(source_directory + "/" + dir_iter->filename(),
+                    string_list sublist = build_pkg_list(source_directory + "/" + dir_iter->path().filename().string(),
                             recursive);
 #endif
                     list.insert(list.end(), sublist.begin(), sublist.end());
@@ -631,7 +631,7 @@
 #if BOOST_VERSION_MAJOR <= 1 and BOOST_VERSION_MINOR < 36
                     std::string name = dir_iter->leaf();
 #else
-                    std::string name = dir_iter->filename();
+                    std::string name = dir_iter->path().filename().string();
 #endif
                     if ((int)name.find(package_suffix) > 0 && (name.length()
                             - name.find(package_suffix) == package_suffix.length()))
