diff --git a/arch/x86/kernel/cpu/microcode/intel.c b/arch/x86/kernel/cpu/microcode/intel.c
index 65cbbcd..611914e 100644
--- a/arch/x86/kernel/cpu/microcode/intel.c
+++ b/arch/x86/kernel/cpu/microcode/intel.c
@@ -549,13 +549,9 @@ scan_microcode(struct mc_saved_data *mcs, unsigned long *mc_ptrs,
 	cd.data = NULL;
 	cd.size = 0;
 
-	/* try built-in microcode if no initrd */
-	if (!size) {
-		if (!load_builtin_intel_microcode(&cd))
-			return UCODE_ERROR;
-	} else {
-		cd = find_cpio_data(p, (void *)start, size, &offset);
-		if (!cd.data)
+	cd = find_cpio_data(p, (void *)start, size, &offset);
+	if (!cd.data) {
+		if (!load_builtin_intel_microcode(&cd))
 			return UCODE_ERROR;
 	}
 
@@ -569,7 +565,7 @@ scan_microcode(struct mc_saved_data *mcs, unsigned long *mc_ptrs,
 static void
 print_ucode_info(struct ucode_cpu_info *uci, unsigned int date)
 {
-	pr_info_once("microcode updated early to revision 0x%x, date = %04x-%02x-%02x\n",
+	pr_info_once("built-in microcode updated early to revision 0x%x, date = %04x-%02x-%02x\n",
 		     uci->cpu_sig.rev,
 		     date & 0xffff,
 		     date >> 24,
