diff -Naur ppp-2.4.7/pppd/plugins/rp-pppoe/pppoe.h ppp-2.4.7-p/pppd/plugins/rp-pppoe/pppoe.h
--- rp-pppoe-3.12/src/pppoe.h	2014-08-09 14:31:39.000000000 +0200
+++ rp-pppoe-3.12/src/pppoe.h	2017-03-06 14:06:02.824393112 +0100
@@ -47,10 +47,8 @@
 #include <sys/socket.h>
 #endif
 
-/* Ugly header files on some Linux boxes... */
-#if defined(HAVE_LINUX_IF_H)
-#include <linux/if.h>
-#elif defined(HAVE_NET_IF_H)
+/* Linux include after glibc includes OR just use *one* interface  */
+#if defined(HAVE_NET_IF_H)
 #include <net/if.h>
 #endif
 
@@ -80,10 +78,6 @@
 #error Could not find a 32-bit integer type
 #endif
 
-#ifdef HAVE_LINUX_IF_ETHER_H
-#include <linux/if_ether.h>
-#endif
-
 #include <netinet/in.h>
 
 #ifdef HAVE_NETINET_IF_ETHER_H
@@ -97,7 +91,13 @@
 #endif
 #endif
 
+#if defined(HAVE_LINUX_IF_H)
+#include <linux/if.h>
+#endif
 
+#ifdef HAVE_LINUX_IF_ETHER_H
+#include <linux/if_ether.h>
+#endif
 
 /* Ethernet frame types according to RFC 2516 */
 #define ETH_PPPOE_DISCOVERY 0x8863
