--- rp-pppoe-3.12/src/configure.orig	2015-11-11 13:56:41.906153374 -0500
+++ rp-pppoe-3.12/src/configure	2015-11-11 13:56:53.749531913 -0500
@@ -3709,6 +3709,7 @@
 #include<net/ethernet.h>
 #include<linux/if.h>
 #include<linux/in.h>
+#include<linux/in6.h>
 
 "
 if test "x$ac_cv_header_linux_if_pppox_h" = xyes; then :
@@ -4558,6 +4559,7 @@
 #include <net/ethernet.h>
 #include <linux/if.h>
 #include <netinet/in.h>
+#include<linux/in6.h>
 #include <linux/if_pppox.h>
 int main()
 {

