--- a/contrib/dracut/90zfs/module-setup.sh.in	2022-06-25 21:27:15.418708560 +0200
+++ b/contrib/dracut/90zfs/module-setup.sh.in	2022-07-03 14:32:34.044234587 +0200
@@ -47,7 +47,7 @@
 			# Otherwise, use dracut's library installation function to find the right one
 		elif ! inst_libdir_file "libgcc_s.so*"; then
 			# If all else fails, just try looking for some gcc arch directory
-			inst_simple /usr/lib/gcc/*/*/libgcc_s.so* ||
+			inst_simple /usr/lib/libgcc_s.so* ||
 				{ dfatal "Unable to install libgcc_s.so"; exit 1; }
 		fi
 	fi

