diff -Naur burg-theme-20100623-orig/themes/icons/grey burg-theme-20100623/themes/icons/grey
--- burg-theme-20100623-orig/themes/icons/grey	2010-05-04 20:32:56.000000000 +0000
+++ burg-theme-20100623/themes/icons/grey	2010-11-25 19:20:49.770001484 +0000
@@ -22,6 +22,7 @@
   -zenix { image = "$$/grey_zenix.png" }
   -molinux { image = "$$/grey_molinux.png" }
   -image { image = "$$/grey_unknown.png" }
+  -chakra { image = "$$/grey_chakra.png" }
 }
 
 -include "../custom/icon_grey"
diff -Naur burg-theme-20100623-orig/themes/icons/hover burg-theme-20100623/themes/icons/hover
--- burg-theme-20100623-orig/themes/icons/hover	2010-05-05 00:07:48.000000000 +0000
+++ burg-theme-20100623/themes/icons/hover	2010-11-25 19:21:12.506668152 +0000
@@ -22,6 +22,7 @@
   -zenix { image = "$$/grey_zenix.png:$$/large_zenix.png" }
   -molinux { image = "$$/grey_molinux.png:$$/large_molinux.png" }
   -image { image = "$$/grey_unknown.png:$$/large_unknown.png" }
+  -chakra { image = "$$/grey_chakra.png:$$/large_chakra.png" }
 }
 
 -include "../custom/icon_hover"
diff -Naur burg-theme-20100623-orig/themes/icons/large burg-theme-20100623/themes/icons/large
--- burg-theme-20100623-orig/themes/icons/large	2010-05-06 02:30:08.000000000 +0000
+++ burg-theme-20100623/themes/icons/large	2010-11-25 19:21:31.963334820 +0000
@@ -22,6 +22,7 @@
   -zenix { image = "$$/large_zenix.png" }
   -molinux { image = "$$/large_molinux.png" }
   -image { image = "$$/large_unknown.png" }
+  -chakra { image = "$$/large_chakra.png" }
 }
 
 -include "../custom/icon_large"
diff -Naur burg-theme-20100623-orig/themes/icons/small burg-theme-20100623/themes/icons/small
--- burg-theme-20100623-orig/themes/icons/small	2010-05-06 02:30:14.000000000 +0000
+++ burg-theme-20100623/themes/icons/small	2010-11-25 19:21:54.213334819 +0000
@@ -22,6 +22,7 @@
   -zenix { image = "$$/small_zenix.png" }
   -molinux { image = "$$/small_molinux.png" }
   -image { image = "$$/small_unknown.png" }
+  -chakra { image = "$$/small_chakra.png" }
 }
 
 -include "../custom/icon_small"
