--- boost_1_65_1/tools/build/src/tools/python.jam
+++ boost_1_65_1/tools/build/src/tools/python.jam
@@ -939,14 +939,14 @@
     # for a particular target OS as the default. This makes it so that we can
     # select a python interpreter with only knowledge of the target OS. And hence
     # can configure different Pythons based on the target OS only.
-    local toolset-requirements = [ toolset.requirements ] ;
-    local toolset-target-os-requirements
-        = [ property.evaluate-conditionals-in-context
-            [ $(toolset-requirements).raw ] : <target-os>$(target-os) ] ;
-    if ! <python> in $(toolset-target-os-requirements:G)
-    {
-        toolset.add-requirements <target-os>$(target-os):<python>$(version:E=default) ;
-    }
+    #local toolset-requirements = [ toolset.requirements ] ;
+    #local toolset-target-os-requirements
+    #    = [ property.evaluate-conditionals-in-context
+    #        [ $(toolset-requirements).raw ] : <target-os>$(target-os) ] ;
+    #if ! <python> in $(toolset-target-os-requirements:G)
+    #{
+    #    toolset.add-requirements <target-os>$(target-os):<python>$(version:E=default) ;
+    #}
 
     # Register the right suffix for extensions.
     register-extension-suffix $(extension-suffix) : $(target-requirements) ;
